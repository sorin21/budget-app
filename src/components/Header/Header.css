.Header {
  margin: 10px 0;
  box-sizing: border-box;
  display: block;
  width: 100%;
  background-color: red;
}

a {
  color: #8F5C2C;
  text-decoration: none;
  padding: 10px 10px;
  box-sizing: border-box;
  display: inline-block;
}

a:hover, a:active, a.active {
  color: #40A4C8;
  display: inline-block;
}