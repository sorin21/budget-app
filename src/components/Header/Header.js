import React, { Component } from 'react';
import { NavLink } from 'react-router-dom'

import classes from './Header.css'


class Header extends Component {
  render() {
    return (
      <header>
        <h1>Income and expenses</h1>
        <NavLink activeClassName={classes.active} to="/" exact>Home</NavLink>
        <NavLink activeClassName={classes.active} to="/create">Add Expense</NavLink>
        <NavLink activeClassName={classes.active} to="/edit">Edit Expense</NavLink>
      </header>
    );
  }
}

export default Header;