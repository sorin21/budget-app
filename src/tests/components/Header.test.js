import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ReactShallowRenderer from 'react-test-renderer/shallow';

// import '../setupTests';
import Header from '../../components/Header/Header';

// enzyme-to-json convert Enzyme wrappers to a format compatible 
// with Jest snapshot testing.
test('should render Header component', () => {
  const wrapper = shallow(<Header />);
  expect(toJSON(wrapper)).toMatchSnapshot();
  // expect(wrapper.find('h1').text()).toBe('Income and expenses')
  // const renderer = new ReactShallowRenderer;
  // renderer.render(<Header />);
  // expect(renderer.getRenderOutput()).toMatchSnapshot()
  // console.log(renderer.getRenderOutput())
});