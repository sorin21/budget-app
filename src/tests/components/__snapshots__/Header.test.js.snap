// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component 1`] = `
<header>
  <h1>
    Income and expenses
  </h1>
  <NavLink
    activeClassName="active"
    exact={true}
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="active"
    to="/create"
  >
    Add Expense
  </NavLink>
  <NavLink
    activeClassName="active"
    to="/edit"
  >
    Edit Expense
  </NavLink>
</header>
`;
